File: sort3.py
"""
ID: totheso1
LANG: PYTHON3
TASK: sort3
"""

from typing import Any, Counter, List
from itertools import islice

fin = open("sort3.in", "r")
fout = open("sort3.out", "w")


def fprint(*args: Any, **kwargs: Any) -> None:
    print(*args, file=fout, **kwargs)


# Code start
MAX_NUMBER = int(fin.readline())
number_collection = tuple(map(int, fin))
number_counter = Counter(number_collection)
section_counter_collection: List[Counter[int]] = [Counter()]
b = 0
e = b
for i in range(1, 4):
    e += number_counter[i]
    section_counter_collection.append(Counter(islice(number_collection, b, e)))
    b += number_counter[i]
answer = 0
for i in range(1, 4):
    for j in range(1, i):
        answer += min(
            section_counter_collection[i][j], section_counter_collection[j][i]
        )
answer += 2 * (
    max(section_counter_collection[1][2], section_counter_collection[2][1])
    - min(section_counter_collection[1][2], section_counter_collection[2][1])
)
fprint(answer)
# Code end

File: sort3.cpp
/*
ID: totheso1
LANG: C++14
TASK: sort3
*/

#include <algorithm>
#include <array>
#include <fstream>
#include <iterator>
#include <vector>

using namespace std;

static ifstream fin("sort3.in");
static ofstream fout("sort3.out");

#define endl '\n'

template <typename T>
T fin_get() {
  return *istream_iterator<T>(fin);
}

template <typename C>
C fin_get_collection(const int size) {
  C ret;
  copy_n(istream_iterator<typename C::value_type>(fin), size,
         back_inserter(ret));
  return ret;
}

int main() {
  constexpr int MAX_NUMBER = 3;
  const int NUMBER_COUNT = fin_get<int>();
  const auto number_collection = fin_get_collection<vector<int>>(NUMBER_COUNT);
  array<int, MAX_NUMBER + 1> count_by_number_collection = {};
  for (const int number : number_collection) {
    ++count_by_number_collection[number];
  }
  array<array<int, MAX_NUMBER + 1>, MAX_NUMBER + 1>
      count_by_position_then_number_collection = {};
  auto b = cbegin(number_collection), e = b;
  for (int position = 1; position <= 3; ++position) {
    e += count_by_number_collection[position];
    for (; b < e; ++b) {
      ++count_by_position_then_number_collection[position][*b];
    }
  }
  int answer = 0;
  for (int i = 1; i <= 3; ++i) {
    for (int j = 1; j < i; ++j) {
      answer += min(count_by_position_then_number_collection[i][j],
                    count_by_position_then_number_collection[j][i]);
    }
  }
  answer += 2 * (max(count_by_position_then_number_collection[1][2],
                     count_by_position_then_number_collection[2][1]) -
                 min(count_by_position_then_number_collection[1][2],
                     count_by_position_then_number_collection[2][1]));
  fout << answer << endl;
}
